lib/httplib/client.go

```diff
@@ -182,6 +182,9 @@ func GetClient(insecure bool, options ...ClientOption) *http.Client {
 	for _, o := range options {
 		o(client)
 	}
+	if client.Timeout == 0 {
+		client.Timeout = defaults.ClientTimeout
```

Comment:
Hmm, I'm not sure there should be a default timeout on the client. Esp. the one as short as 30 seconds. This is overall timeout, not just the connect timeout, right? There are some potentially long-running gravity commands, like downloading packages, which may be timing out because of this?

Reply:
Oh... yea I guess this wouldn't work then.

Reply:
Running without timeout is also not ideal. In case of long-running commands, they would either disable or set a longer timeout explicitly.
@bernardjkim I think setting the default should be done **before** applying the options to let the caller override the defaults.
@r0mant let's discuss that in more detail.

Reply:
@a-palchikov @bernardjkim My proposal is to have a default *connect* timeout and no default overall timeout which is very request-specific.

Reply:
@bernardjkim Ok, so let's set the dialer's timeout to some reasonable default and leave the client's timeout a caller's responsibility for the time being so this gets merged.